implementations/rust/ockam/ockam_node/src/context/transports.rs

```diff
@@ -30,14 +32,26 @@ impl Context {
         for transport in transports.values() {
```

Comment:
Do you think it's possible to simplify some logic considering the fact that there can be only 1 hop resolved here?

Reply:
yes we could check if the returned route has been fully resolved or not. Let me try that

Reply:
For example, there is a check in TCPs `resolve_route` to prevent more than 1 hop, but can't we instead look at the route in this function, find the first transport hop, and then ask the corresponding transport to resolve it?

Reply:
This way, we even add support for a more complex topology and let the intermediate node to resolve its transport routes if required. Maybe in the future, we could also use that logic for secure channels (what is it's a transport type as well?)

Reply:
Nevermind, we don't want more than 1 transport hop in the route.